src/test/scala/com/rbmhtechnology/calliope/javadsl/DeduplicationMapSpec.scala

```diff
@@ -0,0 +1,179 @@
+      override def topicPartition(entry: Entry): TopicPartition = entry.tp
+    }
+  )
+
+  object Entry {
+    def apply(topic: String, partition: Int, aggregateId: String, payload: String): Entry =
+      apply(new TopicPartition(topic, partition), aggregateId, aggregateId + "_" + topic + "_" + partition + "payload-" + payload)
+  }
+
+  case class Entry(tp: TopicPartition, aggregateId: String, payload: String)
+}
+
+class DeduplicationMapSpec extends WordSpecLike with MustMatchers {
+
+  import DeduplicationMapSpec._
+  import JOption._
+
+  "A DeduplicationMap" when {
+    "an entry is appended to the map and" when {
+      "the map is empty" must {
+        "append entry" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+
+          map.size must be(1)
+          map.asScala must contain only Entry(Topic1, 0, "agg1", "p-1")
+        }
+      }
+      "the map contains an entry for the same aggregate in the same partition" must {
+        "discard entry" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic1, 0, "agg1", "p-2"))
+
+          map.asScala must contain only Entry(Topic1, 0, "agg1", "p-1")
+        }
+      }
+      "the map does not contain the aggregate of the entry" must {
+        "append entry" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic1, 0, "agg2", "p-2"))
+
+          map.asScala must contain allOf(Entry(Topic1, 0, "agg1", "p-1"), Entry(Topic1, 0, "agg2", "p-2"))
+        }
+      }
+      "the map does not contain aggregate of the entry in the same partition" must {
+        "append entry" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic1, 1, "agg1", "p-2"))
+
+          map.asScala must contain allOf(Entry(Topic1, 0, "agg1", "p-1"), Entry(Topic1, 1, "agg1", "p-2"))
+        }
+      }
+      "the map does not contain aggregate of the entry in the same topic" must {
+        "append entry" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic2, 0, "agg1", "p-2"))
+
+          map.asScala must contain allOf(Entry(Topic1, 0, "agg1", "p-1"), Entry(Topic2, 0, "agg1", "p-2"))
+        }
+      }
+      "iterated over single partition" must {
+        "keep a stable order per partition" in {
+          val map = EmptyDeduplicationMap
+            .append(Entry(Topic1, 0, "agg1", "p-0"))
+            .append(Entry(Topic1, 0, "agg1", "p-1"))
+            .append(Entry(Topic1, 0, "agg2", "p-2"))
+            .append(Entry(Topic1, 0, "agg2", "p-3"))
+            .append(Entry(Topic1, 0, "agg3", "p-4"))
+            .append(Entry(Topic1, 0, "agg3", "p-5"))
+
+          map.size must be(3)
+          map.asScala.toList must contain inOrderOnly(
```

Comment:
Cant we just say `mustBe List(...)` to get rid of the separate size-check above? (here and below in the other tests)

Reply:
the size-check is performed to check if the resulting-size is calculated correctly in the `DeduplicationMap`

Reply:
I will add a separate test for size-calculation though and remove the size-check in all other tests